tidy3d/plugins/near2far/near2far.py

```diff
@@ -1,130 +1,483 @@
 """Near field to far field transformation plugin
 """
+from typing import List, Dict
 import numpy as np
-
-from ...constants import C_0, ETA_0
-from ...components.data import SimulationData
+import xarray as xr
+import pydantic
+
+from ...constants import C_0, ETA_0, MICROMETER
+from ...components.data import SimulationData, FieldData
+from ...components.monitor import FieldMonitor
+from ...components.types import Direction, Axis, Size
+from ...components.medium import Medium
+from ...components.validators import assert_plane
 from ...log import SetupError
 
-# TODO: implement new version with simulation.discretize
 
+class Near2FarSurface(pydantic.BaseModel):
+    """Data structure to store surface monitor data with associated surface current densities."""
+
+    monitor: FieldMonitor = pydantic.Field(
+        ...,
+        title="Field monitor",
+        description=
+        "Object of :class:`.FieldMonitor` on which near fields will be sampled and integrated.",
+    )
+
+    normal_dir: Direction = pydantic.Field(
+        '+',
```

Comment:
Let's make this a required argument (no default).  to do this, replace `'+'` with `...`